commands/applycmd.go

```diff
@@ -35,25 +40,56 @@ func GetApplyRunner(provider provider.Provider, loader manifestreader.ManifestLo
 type ApplyRunnerWrapper struct {
 	applyRunner *apply.ApplyRunner
 	factory     cmdutil.Factory
+	ioStreams   genericclioptions.IOStreams
+	applyCRD    bool // Apply the ResourceGroup CRD before applying
 }
 
 // Command returns the wrapped ApplyRunner cobraCommand structure.
 func (w *ApplyRunnerWrapper) Command() *cobra.Command {
 	return w.applyRunner.Command
 }
 
-// RunE runs the ResourceGroup CRD installation as a pre-step if an
-// environment variable exists. Then the wrapped ApplyRunner is
-// invoked. Returns an error if one happened. Swallows the
-// "AlreadyExists" error for CRD installation.
+// RunE delegates to the stored applyRunner. Before the delegation, this
+// function either applies the inventory ResourceGroup CRD
+// (--apply-inventory-crd flag), or checks if the inventory
+// ResourceGroup CRD is available. Returns an error if one occurs in the
+// delegation.
 func (w *ApplyRunnerWrapper) RunE(cmd *cobra.Command, args []string) error {
-	if _, exists := os.LookupEnv(resourceGroupEnv); exists {
-		klog.V(4).Infoln("wrapper applyRunner detected environment variable")
-		err := live.ApplyResourceGroupCRD(w.factory)
-		if err != nil && !apierrors.IsAlreadyExists(err) {
-			return err
-		}
+	if w.applyCRD {
+		// Force the apply of the inventory ResourceGroup CRD before applying.
+		// This does nothing (no error) if the CRD already exists.
+		w.ApplyResourceGroupCRD()
+	} else if !live.ResourceGroupCRDApplied(w.factory) {
+		// Otherwise, report the inventory ResourceGroup if missing.
+		fmt.Fprintln(w.ioStreams.Out, "inventory ResourceGroup CRD is missing")
+		fmt.Fprintln(w.ioStreams.Out, "run 'kpt live install-resource-group' to remedy")
```

Comment:
Should there be a return here? Also, should we handle this error through the centralized error handler?

Reply:
Since the `Run` will error out immediately, we can continue or return here. Let me know what you'd like.

Reply:
After running e2e tests, I realized returning here would break legacy `ConfigMap` inventory applies. So we can not return here.

Reply:
That sounds strange. Does that mean that if we use the legacy inventory, kpt will complain about missing ResourceGroup CRD?

Reply:
Yes.

Reply:
Can we fix that? It sounds like a pretty confusing UX.

Reply:
Yes. I was hoping to fix it in a follow-on PR.